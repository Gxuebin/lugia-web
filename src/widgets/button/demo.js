/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import Button from './index';

export default () => {
  return <Button type="danger">hello</Button>;
};
