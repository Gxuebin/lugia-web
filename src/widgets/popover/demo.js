/**
 *
 * create by liangguodong on 2018/12/5
 *
 * @flow
 */
import * as React from 'react';
import Direction from '../button';
import styled from 'styled-components';
import Popover from './popover';

class InnerCloseDemo extends React.Component {
  state = {
    visible: false,
  };

  hide = () => {
    this.setState({
      visible: false,
    });
  };

  handleVisibleChange = visible => {
    this.setState({ visible });
  };

  render() {
    const description = 'this is description';
    return (
      <Popover
        arrowPosition="top"
        title="this is the title"
        action="click"
        visible={this.state.visible}
        onVisibleChange={this.handleVisibleChange}
        clear={'lugia-icon-reminder_close'}
        description={[<div>{description}</div>]}
        onClearClick={this.hide}
      >
        <Direction type="primary">Click me</Direction>
      </Popover>
    );
  }
}
class HoverAndClcikDemo extends React.Component {
  state = {
    clicked: false,
    hovered: false,
  };

  hide = () => {
    this.setState({
      clicked: false,
      hovered: false,
    });
  };

  handleHoverChange = visible => {
    const { onVisibleChange } = this.props;
    this.setState({
      hovered: visible,
      clicked: false,
    });
    onVisibleChange && onVisibleChange(visible);
  };

  handleClickChange = visible => {
    this.setState({
      clicked: visible,
      hovered: false,
    });
  };

  render() {
    const hoverContent = <div>This is hover content.</div>;
    const clickContent = <div>This is click content.</div>;
    return (
      <Popover
        style={{ width: 500 }}
        description={hoverContent}
        title="This is  hover title"
        action={'hover'}
        arrowPosition="topLeft"
        visible={this.state.hovered}
        onVisibleChange={this.handleHoverChange}
      >
        <div>
          <Popover
            description={clickContent}
            title="This is  click title"
            action={'click'}
            arrowPosition="topLeft"
            visible={this.state.clicked}
            clear={'lugia-icon-reminder_close'}
            onVisibleChange={this.handleClickChange}
            onClearClick={this.hide}
          >
            <Direction type="primary">Hover and click / 悬停并单击</Direction>
          </Popover>
        </div>
      </Popover>
    );
  }
}
const Wrapper = styled.div`
  margin: 100px;
`;

export const WrapperDemo = () => {
  const text = 'this is title ';
  const description = 'this is description';
  return (
    <Wrapper>
      <div style={{ marginLeft: 70, whiteSpace: 'nowrap' }}>
        <Popover arrowPosition="topLeft" title={text} action={'click'}>
          <Direction>TL</Direction>
        </Popover>
        <Popover
          arrowPosition="top"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>Top</Direction>
        </Popover>
        <Popover
          arrowPosition="topRight"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>TR</Direction>
        </Popover>
      </div>
      <div style={{ width: 70, float: 'left' }}>
        <Popover
          arrowPosition="leftTop"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>LT</Direction>
        </Popover>
        <Popover
          arrowPosition="left"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>Left</Direction>
        </Popover>
        <Popover
          arrowPosition="leftBottom"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>LB</Direction>
        </Popover>
      </div>
      <div style={{ width: 70, marginLeft: 200 }}>
        <Popover
          arrowPosition="rightTop"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>RT</Direction>
        </Popover>
        <Popover
          arrowPosition="right"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>Right</Direction>
        </Popover>
        <Popover
          arrowPosition="rightBottom"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>RB</Direction>
        </Popover>
      </div>
      <div style={{ marginLeft: 70, clear: 'both', whiteSpace: 'nowrap' }}>
        <Popover
          arrowPosition="bottomLeft"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>BL</Direction>
        </Popover>
        <Popover
          arrowPosition="bottom"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>Bottom</Direction>
        </Popover>
        <Popover
          arrowPosition="bottomRight"
          title={text}
          description={[<div>{description}</div>, <div>{description}</div>]}
        >
          <Direction>BR</Direction>
        </Popover>
      </div>
      <br />
      <Popover
        arrowPosition="bottom"
        title={text}
        action={'focus'}
        description={[<div>{description}</div>, <div>{description}</div>]}
      >
        <Direction>聚焦</Direction>
      </Popover>
      <Popover
        arrowPosition="bottom"
        title={text}
        action={'hover'}
        description={[<div>{description}</div>, <div>{description}</div>]}
      >
        <Direction> 悬停</Direction>
      </Popover>
      <Popover
        arrowPosition="bottom"
        title={text}
        action={'click'}
        arrowPosition="top"
        description={[<div>{description}</div>, <div>{description}</div>]}
      >
        <Direction>点击</Direction>
      </Popover>
      <br />
    </Wrapper>
  );
};
export default () => {
  return [<WrapperDemo />, <HoverAndClcikDemo />, <InnerCloseDemo />];
};
