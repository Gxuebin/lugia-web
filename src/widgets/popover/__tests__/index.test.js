/**
 *
 * create by liangguodong
 *
 * @flow
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { WrapperDemo } from '../demo';
import 'jest-styled-components';

describe('popover Demo', () => {
  it('Component JSON', () => {
    global.svtest = true;
    const render = renderer.create(<WrapperDemo />);
    expect(render.toJSON()).toMatchSnapshot();
    global.svtest = false;
  });
});
