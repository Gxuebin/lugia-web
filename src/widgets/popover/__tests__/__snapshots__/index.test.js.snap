// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popover Demo Component JSON 1`] = `
.c1 {
  display: inline-block;
  margin-bottom: 0;
  text-align: center;
  touch-action: manipulation;
  cursor: pointer;
  white-space: nowrap;
  line-height: 1;
  font-family: Trebuchet Ms,Arial,Helvetica,sans-serif;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  position: relative;
  text-transform: none;
  outline: 0;
  vertical-align: bottom;
  color: #666666;
  background-color: #ffffff;
  border: 1px solid #cccccc;
  height: 2.666666666666667em;
  padding: 0.8333333333333334em;
  font-size: 1em;
  border-radius: 0.33333333333333337em;
}

.c1:hover {
  color: #8090ff;
  border: 1px solid #8090ff;
  background-color: rgba(77,99,255,0.05);
}

.c1:focus {
  color: #8090ff;
  border: 1px solid #8090ff;
  background-color: rgba(77,99,255,0.05);
}

.c1:active {
  color: #3e4fcc;
  border: 1px solid #3e4fcc;
}

.c2 {
  display: none;
}

.c4 {
  position: absolute;
  background-color: #fff;
  box-shadow: 0 1px 6px rgba(0,0,0,0.2);
  border-radius: 4px;
  box-sizing: border-box;
  left: -9999px;
  top: -9999px;
  z-index: 1051;
}

.c3 {
  padding-bottom: 0.8333333333333334em;
  box-shadow: none;
}

.c16 {
  padding-right: 0.8333333333333334em;
  box-shadow: none;
}

.c23 {
  padding-left: 0.8333333333333334em;
  box-shadow: none;
}

.c30 {
  padding-top: 0.8333333333333334em;
  box-shadow: none;
}

.c5 {
  position: relative;
  font-size: 1.2rem;
  line-height: 1;
  color: #ffffff;
  box-sizing: border-box;
  z-index: 2;
}

.c6 {
  left: 0.8333333333333334em;
  bottom: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(225deg);
  -ms-transform: rotateZ(225deg);
  transform: rotateZ(225deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c11 {
  left: 46%;
  bottom: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(225deg);
  -ms-transform: rotateZ(225deg);
  transform: rotateZ(225deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c14 {
  right: 0.8333333333333334em;
  bottom: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(225deg);
  -ms-transform: rotateZ(225deg);
  transform: rotateZ(225deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c17 {
  top: 0.8333333333333334em;
  right: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(135deg);
  -ms-transform: rotateZ(135deg);
  transform: rotateZ(135deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c19 {
  top: 46%;
  right: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(135deg);
  -ms-transform: rotateZ(135deg);
  transform: rotateZ(135deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c21 {
  bottom: 0.8333333333333334em;
  right: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(135deg);
  -ms-transform: rotateZ(135deg);
  transform: rotateZ(135deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c24 {
  top: 0.8333333333333334em;
  left: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(315deg);
  -ms-transform: rotateZ(315deg);
  transform: rotateZ(315deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c26 {
  top: 46%;
  left: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(315deg);
  -ms-transform: rotateZ(315deg);
  transform: rotateZ(315deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c28 {
  bottom: 0.8333333333333334em;
  left: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(315deg);
  -ms-transform: rotateZ(315deg);
  transform: rotateZ(315deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c31 {
  left: 0.8333333333333334em;
  top: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c33 {
  left: 46%;
  top: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c35 {
  right: 0.8333333333333334em;
  top: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
  box-shadow: 0 0 0.5em rgba(0,0,0,0.15);
  z-index: -1;
}

.c7 {
  left: 0.8333333333333334em;
  bottom: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(225deg);
  -ms-transform: rotateZ(225deg);
  transform: rotateZ(225deg);
  z-index: 0;
}

.c12 {
  left: 46%;
  bottom: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(225deg);
  -ms-transform: rotateZ(225deg);
  transform: rotateZ(225deg);
  z-index: 0;
}

.c15 {
  right: 0.8333333333333334em;
  bottom: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(225deg);
  -ms-transform: rotateZ(225deg);
  transform: rotateZ(225deg);
  z-index: 0;
}

.c18 {
  top: 0.8333333333333334em;
  right: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(135deg);
  -ms-transform: rotateZ(135deg);
  transform: rotateZ(135deg);
  z-index: 0;
}

.c20 {
  top: 46%;
  right: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(135deg);
  -ms-transform: rotateZ(135deg);
  transform: rotateZ(135deg);
  z-index: 0;
}

.c22 {
  bottom: 0.8333333333333334em;
  right: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(135deg);
  -ms-transform: rotateZ(135deg);
  transform: rotateZ(135deg);
  z-index: 0;
}

.c25 {
  top: 0.8333333333333334em;
  left: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(315deg);
  -ms-transform: rotateZ(315deg);
  transform: rotateZ(315deg);
  z-index: 0;
}

.c27 {
  top: 46%;
  left: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(315deg);
  -ms-transform: rotateZ(315deg);
  transform: rotateZ(315deg);
  z-index: 0;
}

.c29 {
  bottom: 0.8333333333333334em;
  left: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(315deg);
  -ms-transform: rotateZ(315deg);
  transform: rotateZ(315deg);
  z-index: 0;
}

.c32 {
  left: 0.8333333333333334em;
  top: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
  z-index: 0;
}

.c34 {
  left: 46%;
  top: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
  z-index: 0;
}

.c36 {
  right: 0.8333333333333334em;
  top: -0.33333333333333337em;
  position: absolute;
  border-style: solid;
  border-width: 0.6666666666666667em;
  border-top-left-radius: 0.33333333333333337em;
  border-color: #ffffff transparent transparent #ffffff;
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
  z-index: 0;
}

.c8 {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 1.2rem;
  line-height: 1.5;
  overflow: hidden;
  padding: 0.5em 0.6666666666666667em;
  color: #333333;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #ffffff;
  border-radius: 0.33333333333333337em;
  box-shadow: 0 0.16666666666666669em 0.6666666666666667em rgba(0,0,0,0.15);
}

.c10 {
  text-align: inherit;
  white-space: nowrap;
  overflow: hidden;
  display: inline-block;
  font-size: 1.6rem;
  font-weight: 500;
  color: #333333;
}

.c13 {
  text-align: inherit;
  white-space: nowrap;
  overflow: hidden;
  font-size: 1.4rem;
  color: #666666;
}

.c9 {
  font-size: 1.2rem;
  display: inline-block;
  padding: 0.5em 0.6666666666666667em;
}

.c0 {
  margin: 100px;
}

<div
  className="c0"
>
  <div
    style={
      Object {
        "marginLeft": 70,
        "whiteSpace": "nowrap",
      }
    }
  >
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        TL
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c3 c4"
      >
        <div
          className="c5"
          fx="bottom"
        >
          <div
            className="c6"
            fx="bottom"
          />
          <div
            className="c7"
            fx="bottom"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        Top
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c3 c4"
      >
        <div
          className="c5"
          fx="bottom"
        >
          <div
            className="c11"
            fx="bottom"
          />
          <div
            className="c12"
            fx="bottom"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        TR
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c3 c4"
      >
        <div
          className="c5"
          fx="bottom"
        >
          <div
            className="c14"
            fx="bottom"
          />
          <div
            className="c15"
            fx="bottom"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "float": "left",
        "width": 70,
      }
    }
  >
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        LT
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c16 c4"
      >
        <div
          className="c5"
          fx="right"
        >
          <div
            className="c17"
            fx="right"
          />
          <div
            className="c18"
            fx="right"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        Left
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c16 c4"
      >
        <div
          className="c5"
          fx="right"
        >
          <div
            className="c19"
            fx="right"
          />
          <div
            className="c20"
            fx="right"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        LB
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c16 c4"
      >
        <div
          className="c5"
          fx="right"
        >
          <div
            className="c21"
            fx="right"
          />
          <div
            className="c22"
            fx="right"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginLeft": 200,
        "width": 70,
      }
    }
  >
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        RT
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c23 c4"
      >
        <div
          className="c5"
          fx="left"
        >
          <div
            className="c24"
            fx="left"
          />
          <div
            className="c25"
            fx="left"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        Right
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c23 c4"
      >
        <div
          className="c5"
          fx="left"
        >
          <div
            className="c26"
            fx="left"
          />
          <div
            className="c27"
            fx="left"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        RB
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c23 c4"
      >
        <div
          className="c5"
          fx="left"
        >
          <div
            className="c28"
            fx="left"
          />
          <div
            className="c29"
            fx="left"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "clear": "both",
        "marginLeft": 70,
        "whiteSpace": "nowrap",
      }
    }
  >
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        BL
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c30 c4"
      >
        <div
          className="c5"
          fx="top"
        >
          <div
            className="c31"
            fx="top"
          />
          <div
            className="c32"
            fx="top"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        Bottom
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c30 c4"
      >
        <div
          className="c5"
          fx="top"
        >
          <div
            className="c33"
            fx="top"
          />
          <div
            className="c34"
            fx="top"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
    >
      <span>
        BR
      </span>
    </button>
    <div
      className="c2"
    >
      <div
        className="c30 c4"
      >
        <div
          className="c5"
          fx="top"
        >
          <div
            className="c35"
            fx="top"
          />
          <div
            className="c36"
            fx="top"
          />
          <div
            className="c8"
          >
            <div
              className="c9"
            >
              <div
                className="c10"
              >
                this is title 
                 
              </div>
              <div
                className="c13"
              >
                <div>
                  this is description
                </div>
                <div>
                  this is description
                </div>
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <br />
  <button
    className="c1"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
  >
    <span>
      聚焦
    </span>
  </button>
  <div
    className="c2"
  >
    <div
      className="c30 c4"
    >
      <div
        className="c5"
        fx="top"
      >
        <div
          className="c33"
          fx="top"
        />
        <div
          className="c34"
          fx="top"
        />
        <div
          className="c8"
        >
          <div
            className="c9"
          >
            <div
              className="c10"
            >
              this is title 
               
            </div>
            <div
              className="c13"
            >
              <div>
                this is description
              </div>
              <div>
                this is description
              </div>
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="c1"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
  >
    <span>
       悬停
    </span>
  </button>
  <div
    className="c2"
  >
    <div
      className="c30 c4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="c5"
        fx="top"
      >
        <div
          className="c33"
          fx="top"
        />
        <div
          className="c34"
          fx="top"
        />
        <div
          className="c8"
        >
          <div
            className="c9"
          >
            <div
              className="c10"
            >
              this is title 
               
            </div>
            <div
              className="c13"
            >
              <div>
                this is description
              </div>
              <div>
                this is description
              </div>
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="c1"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
  >
    <span>
      点击
    </span>
  </button>
  <div
    className="c2"
  >
    <div
      className="c3 c4"
    >
      <div
        className="c5"
        fx="bottom"
      >
        <div
          className="c11"
          fx="bottom"
        />
        <div
          className="c12"
          fx="bottom"
        />
        <div
          className="c8"
        >
          <div
            className="c9"
          >
            <div
              className="c10"
            >
              this is title 
               
            </div>
            <div
              className="c13"
            >
              <div>
                this is description
              </div>
              <div>
                this is description
              </div>
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <br />
</div>
`;
